third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -310,6 +310,30 @@ fn parse_identifier(context: &mut Context) -> Result<Name, Box<Diagnostic>> {
     Ok(spanned(context.tokens.file_hash(), start_loc, end_loc, id))
 }
 
+// Parse an identifier or an anonymous field
+//     <Identifier> | (0-9)*
+fn parse_identifier_or_anonymous_field(context: &mut Context) -> Result<Name, Box<Diagnostic>> {
+    let start_loc = context.tokens.start_loc();
+    let id: Symbol = context.tokens.content().into();
+    if (context.tokens.peek() != Tok::Identifier && context.tokens.peek() != Tok::NumValue)
+        || (context.tokens.peek() == Tok::NumValue
+            && id.as_str().chars().any(|c| !c.is_ascii_digit()))
+    {
+        return Err(unexpected_token_error(
+            context.tokens,
+            "an identifier or an anonymous field `0`, `1`, ...",
```

Comment:
If we're not in Move 2, this should be "an identifier".  Check `context.env.flags().lang_v2()` and use the appropriate message.

Reply:
done